docs/src/development/development.md

```diff
@@ -0,0 +1,92 @@
+# Developer Guide
+
+## Install prerequisites
+
+1. Install [go][go]
+    - Get the latest patch version for go v1.17.
+1. Install [KIND][kind]
+    - `GO111MODULE="on" go get sigs.k8s.io/kind@v<stableVersion>`.
+1. Install [Kustomize][kustomize]
+    - `brew install kustomize` on macOS.
+    - `choco install kustomize` on Windows.
+    - [install instructions][kustomizelinux] on Linux
+1. Install [envsubst][envsubst]
+1. Install make.
+1. Install direnv
+    - `brew install direnv` on macOS.
+
+## Fork and get the source
+
+Fork the [cluster-api-provider-oci repo](https://github.com/oracle/cluster-api-provider-oci):
+
+```bash
+cd "$(go env GOPATH)"/src
+mkdir sigs.k8s.io
+cd sigs.k8s.io/
+git clone git@github.com:<GITHUB USERNAME>/cluster-api-provider-oci.git
+cd cluster-api-provider-oci
+git remote add upstream git@github.com:orale/cluster-api-provider-oci.git
+git fetch upstream
+```
+
+## Running local management cluster for development
+
+The simplest way to test the code is to run it from local.
+If you have `capoci-controller-manager` running in your management cluster,
+please scale down the deployment:
+
+```bash
+kubectl scale deployment/capoci-controller-manager  --replicas=0 -n cluster-api-provider-oci-system
+```
+
+Create and modify the auth-config.yaml file:
+
+```bash
+cp <repo-path>/hack/auth-config-template.yaml <repo-path>/auth-config.yaml
+```
+
+Then modify the file with your information.
+
+Then run the following commands:
+
+```bash
+export AUTH_CONFIG_DIR="<repo-path>/auth-config.yaml"
+make run
+```
+
+The above step will run the code locally using your local management cluster.
+
+If you want to run your changes using a docker built image execute the
+following steps:
+
+```bash
+export TAG=<tag>
+export REGISTRY="<region-key>.ocir.io/<namespace>"
+make docker-build
+```
+
+`region-key` for example, phx. See the
+[Availability by Region](https://docs.oracle.com/en-us/iaas/Content/Registry/Concepts/registryprerequisites.htm#Availab)
+topic in the Oracle Cloud Infrastructure Registry documentation.
+
+`namespace` is the auto-generated Object Storage namespace string of the tenancy
+(as shown on the Tenancy Information page) that owns the repository to which you
+want to push the image.
+
+Push the resulting docker image to the repository. For more info on how to push
+to OCIR see
+<https://www.oracle.com/webfolder/technetwork/tutorials/obe/oci/registry/index.html>
+
+Execute the following steps to install the image
+
+```bash
+make release-manifests
+kubectl apply -f out/infrastructure-oci/v0.1.0-alpha1/infrastructure-components.yaml
```

Comment:
maybe this should be a different version?

Reply:
The current release is 0.1.0 so maybe just change it to a patch release or something - as long as it's not the same as an available release